runtime/spark/conf/spark/spark-defaults.conf

```diff
@@ -24,7 +24,7 @@ spark.history.fs.logDirectory {%spark.eventLog.dir%}
 spark.history.ui.port 18080
 spark.history.fs.cleaner.enabled false
 #spark.history.store.path {%spark.history.store.path%}
-spark.yarn.historyServer.address ${hadoopconf-yarn.resourcemanager.hostname}:18080
```

Comment:
This line has no problem. 

Refer: https://spark.apache.org/docs/3.2.1/running-on-yarn.html